state-sync/src/coordinator.rs

```diff
@@ -1103,22 +1097,37 @@ impl<T: ExecutorProxyTrait> StateSyncCoordinator<T> {
             self.local_state.trusted_epoch()
         };
         self.local_state.verify_ledger_info(&response_li)?;
-        if let Some(li) = pending_li {
-            if li != response_li {
-                self.local_state.verify_ledger_info(&li)?;
+        if let Some(new_highest_li) = new_highest_li.clone() {
+            if new_highest_li != response_li {
+                self.local_state.verify_ledger_info(&new_highest_li)?;
             }
-            self.pending_ledger_infos.add_li(li);
         }
         self.validate_and_store_chunk(txn_list_with_proof, response_li, None)?;
 
-        // need to sync with local storage to see whether response LI was actually committed
-        // and update pending_ledger_infos accordingly
+        // Need to sync with local storage to update synced version
         self.sync_state_with_local_storage()?;
-        let new_version = self.local_state.synced_version();
+        let synced_version = self.local_state.synced_version();
+
+        // Check if we've synced beyond our current highest known ledger info
+        if let Some(highest_known_ledger_info) = &self.highest_known_ledger_info {
```

Comment:
this is really current target not highest -- it was only the highest when we pegged it.

Reply:
Yes, but that's the trade-off we make. We can't keep updating the highest because then we don't incremental commit :P We have to peg and sit until we've synced

Reply:
Oh I'm suggesting we rename it to current target or just target

Reply:
Done, thanks!